### ISSUE TITLE
Support `inputValueDeprecation` in introspection query

### ISSUE THREAD

**Issue by mdoury-oef** [0no-co/gql.tada#142]
### Describe the bug

The introspection query does not support any configuration and comes with some presets. 
I gave it a quick look, and it seems `gql.tada` uses the `getIntrospectionQuery` method from the `graphql` package to fetch remote schemas ([internally](https://github.com/0no-co/gql.tada/blob/e8820360ae29af899dc1b922c55839bbf4d3eaf2/packages/cli-utils/src/tada.ts#L124-L129) and [when using the CLI](https://github.com/0no-co/gql.tada/blob/e8820360ae29af899dc1b922c55839bbf4d3eaf2/packages/cli-utils/src/index.ts#L40-L46)).

```typescript
getIntrospectionQuery({
    descriptions: true,
    schemaDescription: false,
    inputValueDeprecation: false,
    directiveIsRepeatable: false,
    specifiedByUrl: false,
})
```

As a consequence, when an API introduces a non-breaking change by tagging an input field as deprecated, this yields a TypeScript error even though the input field is still accounted for at runtime. 

Would it be possible to allow configuration of the introspection query?

### gql.tada version

gql.tada v1.3.6

### Validations

- [X] I can confirm that this is a bug report, and not a feature request, RFC, question, or discussion, for which [GitHub Discussions](https://github.com/0no-co/gql.tada/discussions) should be used
- [X] Read the [docs](https://gql-tada.0no.co/).
- [X] Follow our [Code of Conduct](https://github.com/0no-co/gql.tada/blob/main/CODE_OF_CONDUCT.md)

**Comment by kitten** [comment]
This is not 100% resolved yet as GraphQLSP will need to be updated.

**Comment by mdoury-oef** [comment]
Thank you so much @kitten 🙌

**Comment by kitten** [comment]
I noticed that I forgot to actually say what we changed here 😅 Sorry!

For reference, the new schema loaders reimplement `getIntrospectionQuery` entirely to make them a little more maintainable and separate us from changes the `graphql.js` ref implementation may make. We now run a pre-introspection “support” query that checks which features the GraphQL API supports (and account for that query failing too). We then run the appropriate introspection query, pulling down the new features that some older GraphQL APIs may not support dynamically
